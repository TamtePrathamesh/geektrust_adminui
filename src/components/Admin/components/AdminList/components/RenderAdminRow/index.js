import React, { useState, useRef } from 'react';
import PropTypes from 'prop-types';
import AdminActions from '../AdminActions';
import { WithBoundary } from '../../../../../common';
import CheckIcon from '../../../../../images/check.png';
import RemoveIcon from '../../../../../images/remove.png';

import Style from './style.module.css';

const RenderAdminRow = ({
  admin: { id, name, email, role, selection = false },
  onDelete,
  onEdit,
  onSelectOrDeselect,
}) => {
  const [isEdit, setIsEdit] = useState(false);
  const nameRef = useRef(null);
  const emailRef = useRef(null);
  const roleRef = useRef(null);

  return (
    <>
      <tr data-testid="tr" className={selection ? Style.selectedRow : ''}>
        {!isEdit ? (
          <>
            <td>
              <input
                data-testid="selectRow"
                type="checkbox"
                checked={selection}
                onChange={(e) => onSelectOrDeselect(id, e)}
              />
            </td>
            <td className={Style.columnName}>{name}</td>
            <td className={Style.columnName}>{email}</td>
            <td className={Style.columnName}>{role}</td>
            <td>
              <AdminActions onDelete={onDelete} onEditMode={setIsEdit} id={id} />
            </td>
          </>
        ) : (
          <>
            <td>
              <input type="checkbox" checked={selection} onChange={(e) => onSelectOrDeselect(id, e)} />
            </td>
            <td>
              <input type="text" defaultValue={name} ref={nameRef} />
            </td>
            <td>
              <input type="text" defaultValue={email} ref={emailRef} />
            </td>
            <td>
              <input type="text" defaultValue={role} ref={roleRef} />
            </td>
            <td>
              <div data-testid="rowEditAction" className={Style.rowEditAction}>
                <img
                  src={CheckIcon}
                  width="24px"
                  height="24px"
                  data-testid="saveRow"
                  onClick={() => {
                    onEdit(id, {
                      name: nameRef.current.value,
                      email: emailRef.current.value,
                      role: roleRef.current.value,
                    });
                    setIsEdit(false);
                  }}
                />
                <img src={RemoveIcon} onClick={() => setIsEdit(false)} width="24px" height="24px" />
              </div>
            </td>
          </>
        )}
      </tr>
    </>
  );
};

RenderAdminRow.propTypes = {
  onDelete: PropTypes.func,
  onEdit: PropTypes.func,
  onSelectOrDeselect: PropTypes.func,
  admin: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    email: PropTypes.string,
    role: PropTypes.string,
    selection: PropTypes.bool,
  }),
};

export default WithBoundary(React.memo(RenderAdminRow));
