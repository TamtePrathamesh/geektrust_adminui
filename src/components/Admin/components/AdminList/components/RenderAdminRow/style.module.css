@import '../../../../../../style/colors.css';

.selectedRow {
    background-color:var(--grey_three);
}

.rowEditAction {
    display: flex;
    align-items: center;
    gap: 15px;
}

img {
    cursor: pointer;
}
