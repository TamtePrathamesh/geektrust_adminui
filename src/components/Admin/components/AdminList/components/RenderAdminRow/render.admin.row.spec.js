import React from 'react';
import { screen, render } from '@testing-library/react';
import user from '@testing-library/user-event';
import RenderAdminRow from '.';

describe('RenderAdminRow', () => {
  it('should match snapshot', () => {
    const onDelete = jest.fn();
    const onEdit = jest.fn();
    const onSelectOrDeselect = jest.fn();
    const tableRow = document.createElement('tbody');
    expect(
      render(
        <RenderAdminRow
          admin={{ id: '1', name: 'xyz', email: 'an@g.c', role: 'member', selection: false }}
          onDelete={onDelete}
          onEdit={onEdit}
          onSelectOrDeselect={onSelectOrDeselect}
        />,
        {
          container: document.body.appendChild(tableRow),
        }
      ).container
    ).toMatchSnapshot();
  });

  it('should display a row in editMode when editClicked', async () => {
    const onDelete = jest.fn();
    const onEdit = jest.fn();
    const onSelectOrDeselect = jest.fn();
    const tableRow = document.createElement('tbody');
    render(
      <RenderAdminRow
        admin={{ id: '1', name: 'xyz', email: 'an@g.c', role: 'member', selection: false }}
        onDelete={onDelete}
        onEdit={onEdit}
        onSelectOrDeselect={onSelectOrDeselect}
      />,
      {
        container: document.body.appendChild(tableRow),
      }
    );
    expect(screen.queryAllByTestId('rowEditAction').length).toBe(0);
    await user.click(screen.queryByTestId('editRow'));
    expect(screen.queryAllByTestId('rowEditAction').length).toBe(1);
    expect(onEdit).toHaveBeenCalledTimes(0);
    await user.click(screen.queryByTestId('saveRow'));
    expect(onEdit).toHaveBeenCalledTimes(1);
  });

  it('should delete a row when delete clicked', async () => {
    const onDelete = jest.fn();
    const onEdit = jest.fn();
    const onSelectOrDeselect = jest.fn();
    const tableRow = document.createElement('tbody');
    render(
      <RenderAdminRow
        admin={{ id: '1', name: 'xyz', email: 'an@g.c', role: 'member', selection: false }}
        onDelete={onDelete}
        onEdit={onEdit}
        onSelectOrDeselect={onSelectOrDeselect}
      />,
      {
        container: document.body.appendChild(tableRow),
      }
    );
    expect(onDelete).toHaveBeenCalledTimes(0);
    await user.click(screen.queryByTestId('deleteRow'));
    expect(onDelete).toHaveBeenCalledTimes(1);
  });

  it('should display selected row as gray color background', async () => {
    const onDelete = jest.fn();
    const onEdit = jest.fn();
    const onSelectOrDeselect = jest.fn();
    const tableRow = document.createElement('tbody');
    render(
      <RenderAdminRow
        admin={{ id: '1', name: 'xyz', email: 'an@g.c', role: 'member', selection: true }}
        onDelete={onDelete}
        onEdit={onEdit}
        onSelectOrDeselect={onSelectOrDeselect}
      />,
      {
        container: document.body.appendChild(tableRow),
      }
    );
    expect(screen.queryByTestId('tr')).toHaveClass('selectedRow');
  });
});
