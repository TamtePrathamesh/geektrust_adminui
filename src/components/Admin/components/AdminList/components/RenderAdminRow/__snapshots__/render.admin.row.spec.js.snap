// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderAdminRow should match snapshot 1`] = `
<tbody>
  <tr
    class=""
    data-testid="tr"
  >
    <td>
      <input
        data-testid="selectRow"
        type="checkbox"
      />
    </td>
    <td
      class="columnName"
    >
      xyz
    </td>
    <td
      class="columnName"
    >
      an@g.c
    </td>
    <td
      class="columnName"
    >
      member
    </td>
    <td>
      <div
        class="actionItems"
      >
        <img
          alt="cant display"
          class="actionIcons"
          data-testid="editRow"
          src=""
        />
        <img
          alt="cant display"
          class="actionIcons"
          data-testid="deleteRow"
          src=""
        />
      </div>
    </td>
  </tr>
</tbody>
`;
